### PR TITLE
✨Update CAPI to v0.4.0

### LINKED ISSUE #918: Update CAPI to v0.4.0
/kind feature

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by hidekazuna**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #918 

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**TODOs**:

- [ ] squashed commits
- if necessary:
  - [ ] includes documentation
  - [ ] adds unit tests

/hold

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -76,8 +76,6 @@ ClusterAPI provider OpenStack images and manifests are published after every PR
 * With a Google Cloud account you can get a quick overview [here](https://console.cloud.google.com/storage/browser/artifacts.k8s-staging-capi-openstack.appspot.com/components)
 * The manifests are available under:
   * [master/infrastructure-components.yaml](https://storage.googleapis.com/artifacts.k8s-staging-capi-openstack.appspot.com/components/master/infrastructure-components.yaml): 
-    latest build from the main branch, overwritten after every merge
```
Comment (sbueringer) [inline README.md:79]: I think we should keep this:
1. the doc is still valid as the nightly build still exists
2. I think we should keep the nightly build as it's still useful

File: test/e2e/data/e2e_conf.yaml [hunk 2]
```diff
@@ -12,12 +12,6 @@
 managementClusterName: capo-e2e
 
 images:
-- name: gcr.io/k8s-staging-cluster-api/cluster-api-controller:nightly_master_20210608
```
Comment (sbueringer) [inline test/e2e/data/e2e_conf.yaml:15]: Please keep those images and just change the tag to v0.4.0.
The idea is that the e2e tests download and import the cache into kind at the start of the e2e tests. This still makes sense with the v0.4.0 images

**Review by chrischdi — COMMENTED** [review]
I agree with the images to @sbueringer :-) 

approve

For the go.mod and go.sum changes :-)

**Comment by sbueringer** [comment]
@hidekazuna  lgtm apart from my two comments. Thx for tackling this! :)

File: README.md (hunk 1 above)
Comment (hidekazuna) [reply README.md:79]: Fixed.

File: test/e2e/data/e2e_conf.yaml (hunk 2 above)
Comment (hidekazuna) [reply test/e2e/data/e2e_conf.yaml:15]: Updated the tag to v0.4.0.

File: README.md [hunk 3]
```diff
@@ -77,8 +77,7 @@ ClusterAPI provider OpenStack images and manifests are published after every PR
 * The manifests are available under:
   * [master/infrastructure-components.yaml](https://storage.googleapis.com/artifacts.k8s-staging-capi-openstack.appspot.com/components/master/infrastructure-components.yaml): 
     latest build from the main branch, overwritten after every merge
-  * e.g. [nightly_master_20210407/infrastructure-components.yaml](https://storage.googleapis.com/artifacts.k8s-staging-capi-openstack.appspot.com/components/nightly_master_20210407/infrastructure-components.yaml): build of the main branch from 7th April
```
Comment (sbueringer) [inline README.md:80]: @hidekazuna looks like it's still deleted

Comment (sbueringer) [reply README.md:80]: You can directly squash after this fix if you want (imho)

**Comment by sbueringer** [comment]
Thx
/approve
/lgtm

**Pull request merged**